lib/src/connection.dart

```diff
@@ -70,18 +81,34 @@ class Connection {
   }
 
   Future<void> connect() async {
-    _socket = await WebSocket.connect(url);
-    _socket?.listen(onMessage);
-    sendPing();
-    _resetCheckPong();
+    try {
+      _socket = await WebSocket.connect(url);
+      _socket?.listen(onMessage);
+      sendPing();
+      _resetCheckPong();
+
+      afterConnect();
+    } catch (error) {
+      print('Connection: connection error $error');
```

Comment:
consider replace it with `developer.log()`

Reply:
I will do it in another PR :)